mods/webui/src/stories/cards/workspace/Workspace.stories.tsx

```diff
@@ -0,0 +1,140 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Meta, StoryObj } from "@storybook/react";
+import { WorkspaceCard } from "./Workspace";
+import { fn } from "@storybook/test";
+
+/**
+ * This story is for the WorkspaceCard component based on Material UI.
+ * It supports both regular and empty variants and can be disabled.
+ */
+const meta = {
+  title: "Shared Components/WorkspaceCard",
+  component: WorkspaceCard,
+  parameters: {
+    layout: "centered",
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/design/OsZlne0RvIgoFlFKF7hnAU/Shared-Component-Library?node-id=8-8505&p=f&t=NCJIzjsjMFiDAc1s-0"
+    }
+  },
+  tags: ["autodocs"],
+  args: { onClick: fn() },
+  argTypes: {
+    onClick: {
+      name: "On Click",
+      description: "Function to execute on click"
+    },
+    onSettingsClick: {
+      name: "On Settings Click",
+      description: "Function to execute on settings click"
+    },
+    workspaceRef: {
+      name: "Workspace Ref",
+      description: "The ref of the workspace card",
+      control: "text",
+      defaultValue: { summary: "workspaceRef" }
+    },
+    region: {
+      name: "Region",
+      description: "The region label of the card",
+      control: "text",
+      defaultValue: { summary: "Region" }
+    },
+    description: {
+      name: "Description",
+      description: "The description of the workspace",
+      control: "text",
+      defaultValue: { summary: "Workspace description" }
+    },
+    date: {
+      name: "Date",
+      description: "The date associated with the workspace",
+      control: "text",
+      defaultValue: { summary: "4/14/24" }
+    },
+    variant: {
+      name: "Variant",
+      description: "The variant to use",
+      options: ["regular", "empty"],
+      control: "radio",
+      defaultValue: { summary: "regular" }
+    },
+    disabled: {
+      name: "Disabled",
+      description: "If true, the card will be disabled",
+      control: "boolean",
+      defaultValue: { summary: false }
+    }
+  }
+} satisfies Meta<typeof WorkspaceCard>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+/**
+ * Example of a regular WorkspaceCard with region, description, and date
+ */
+export const RegularCard: Story = {
+  args: {
+    region: "Region",
+    description: "Demo Workspace With Wrapping Title.",
+    date: "4/14/24",
+    variant: "regular",
+    disabled: false,
+    onSettingsClick: fn(),
+    onClick: fn()
+  }
+};
+
+/**
+ * Example of a regular WorkspaceCard that is disabled
+ */
+export const DisabledCard: Story = {
+  args: {
+    region: "Region",
+    description: "Demo Workspace With Wrapping Title.",
+    date: "4/14/24",
+    variant: "regular",
+    disabled: true
+  }
+};
+
+/**
+ * Example of an empty WorkspaceCard
+ */
+export const EmptyCard: Story = {
+  args: {
+    variant: "empty",
+    disabled: false,
+    onClick: fn()
```

Comment:
There is no need to define the function here as it was already described in the `args` section.